engine/lib/import_thir.ml

```diff
@@ -1440,26 +1440,49 @@ let cast_of_enum typ_name generics typ thir_span
   in
   { v; span; ident; attrs = [] }
 
-let rec c_item ~ident ~drop_body (item : Thir.item) : item list =
+let rec c_item ~ident ~type_only (item : Thir.item) : item list =
   try
     Span.with_owner_hint item.owner_id (fun _ ->
-        c_item_unwrapped ~ident ~drop_body item)
+        c_item_unwrapped ~ident ~type_only item)
   with Diagnostics.SpanFreeError.Exn payload ->
     let context, kind = Diagnostics.SpanFreeError.payload payload in
     let error = Diagnostics.pretty_print_context_kind context kind in
     let span = Span.of_thir item.span in
     [ make_hax_error_item span ident error ]
 
-and c_item_unwrapped ~ident ~drop_body (item : Thir.item) : item list =
+and c_item_unwrapped ~ident ~type_only (item : Thir.item) : item list =
   let open (val make ~krate:item.owner_id.contents.value.krate : EXPR) in
   if should_skip item.attributes then []
   else
     let span = Span.of_thir item.span in
     let attrs = c_item_attrs item.attributes in
+    let erased_by_user =
+      Attr_payloads.payloads attrs
+      |> List.exists ~f:(fst >> [%matches? (Erased : Types.ha_payload)])
+    in
+    let erased_by_type_only =
+      type_only
+      &&
+      match item.kind with
+      | Fn _ | Static _ -> true
+      | Impl { of_trait = Some _; items; _ }
+        when List.exists items ~f:(fun item ->
+                 match item.kind with Type _ -> true | _ -> false)
+             |> not ->
+          true
+      | _ -> false
+    in
+    let attrs =
+      if erased_by_type_only && not erased_by_user then
+        Attr_payloads.to_attr Erased span :: attrs
+      else attrs
+    in
+    let erased = erased_by_user || erased_by_type_only in
```

Comment:
Then, can you add documentation explaining roughly what is the logic, on bindings `erased_by_user`, `erased_by_type_only` and `erased`?
Similarly, please document the `if erased_by_type_only && not erased_by_user then`: the logic is if `erased` is true, then we add a attribute `erased` unless it already exists, right?

Reply:
Yes that's right. I'll add comments.